src/types/utilities.ts

```diff
@@ -6,3 +6,63 @@ export type FArguments<F> = F extends (...args: infer ARGS) => any
   : F extends new (...args: infer ARGS) => any
   ? ARGS
   : never;
+
+/**
+ * Pick a sparse type via a two-parameter path
+ *
+ * @example
+ *
+ * class Foo {
+ *   bar = {
+ *      baz: Promise.resolve(6)
+ *   }
+ *   a = 4;
+ *   b = { c: 'This is c'}
+ * }
+ *
+ * const pickTwoFoo: Pick2<Foo, 'b', 'c'> = new Foo();
+ * pickTwoFoo.b.c; // ☑️ will type-check
+ * pickTwoFoo.bar; // ❌ will NOT type-check
+ *
+ * const otherPickTwoFoo: Pick2<Foo, 'b', 'c'> = new Foo();
+ * otherPickTwoFoo.b; // ❌ will NOT type-check
+ * otherPickTwoFoo.bar.baz; // ☑️ will type-check
+ */
+export type Pick2<
+  OBJ,
```

Comment:
Users get slightly nicer diagnostics if we make this `OBJ extends object`.

Given:

```ts
type Nonsense = Pick<42, 'b', 'c'>;
```

Without the suggested change, red squiggles are on `'b'`, and the error message is:

> Type '"b"' does not satisfy the constraint '"toString" | "valueOf" | "toFixed" | "toExponential" | "toPrecision" | "toLocaleString"'.

With it, red squiggles are on `42` and the error message is:

> Type '42' does not satisfy the constraint 'object'.

However, this isn't actually viable, because we would have to add some additional (conditional) types to make `Pick3` work. As I played with that, I didn't think it was probably worth it (performance and complexity-wise both), but it might be worth keeping an eye on for these kinds of types.